app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/snippet/SnippetDetailsViewModel.kt

```diff
@@ -0,0 +1,184 @@
+    }
+
+    private fun startWatchingSnippet() = launchIO {
+        repo.startWatchingSnippet(workspaceId.value, encodedId.value).handlingErrors(R.string.error_changing_watching) { isUserWatchingSnippet() }
+    }
+
+    // TODO: Show dialog to confirm user wants to continue with deletion before calling this function
+    fun onDeleteSnippetClick() = launchIO {
+        repo.deleteSnippet(workspaceId.value, encodedId.value).handlingErrors(R.string.delete_snippet_error) {
+            // Using handleError here to inform the user that the snippet has been deleted; maybe use a branded messaging?
+            handleError(R.string.delete_snippet_success)
+        }
+    }
+
+    private fun createSnippetComment() = launchIO {
+        repo.createSnippetComment(workspaceId.value, encodedId.value, newSnippetComment.value).handlingErrors(R.string.create_comment_error) {
+            getSnippetComments()
+            clearCommentValues()
+        }
+    }
+
+    private fun createReplyComment(commentId: Int) = launchIO {
+        repo.createCommentReply(workspaceId.value, encodedId.value, newReplyComment.value, commentId).handlingErrors(R.string.comment_reply_error) {
+            getSnippetComments()
+            clearCommentValues()
+        }
+    }
+
+    fun commentEditClick(commentId: Int) = launchIO {
+        repo.editSnippetComment(workspaceId.value, encodedId.value, newSnippetComment.value, commentId).handlingErrors(R.string.edit_comment_error) { getSnippetComments() }
+    }
+
+    fun commentDeleteClick(commentId: Int) = launchIO {
+        repo.deleteSnippetComment(workspaceId.value, encodedId.value, commentId).handlingErrors(R.string.delete_comment_error) { getSnippetComments() }
+    }
+
+    /////////////////////  Callbacks /////////////////////
+    fun changeSnippetWatching() = when (isWatchingSnippet.value) {
+        true -> stopWatchingSnippet()
+        false -> startWatchingSnippet()
+    }
+
+    fun onEditSnippetClick() {}
+
+    fun onCommentSaveEvent(commentId: Int?) {
+        when (commentId == null) {
+            true -> createSnippetComment()
+            false -> createReplyComment(commentId)
+        }
+    }
+
+    /////////////////////  Helper Functions /////////////////////
+    fun clearCommentValues() {
+        newSnippetComment.value = ""
+        newReplyComment.value = ""
+    }
+
+    private fun sortComments(comments: List<SnippetComment>) {
+        val (parentComments, childrenComments) = comments.partition { it.parentId == null }
+
+        childrenComments.forEach { child ->
+            childrenComments.find { it.id == child.parentId }?.childrenComments?.add(child) ?: run {
+                parentComments.find { it.id == child.parentId }?.childrenComments?.add(child)
+            }
+        }
+
+        snippetComments.value = parentComments.reversed()
+    }
+
+    /** Functions for optimized sort theory */
```

Comment:
I'll see about getting this sort working as well as the original and then we can do a short presentation before a code n tell.